commit/src/pcs.rs

```diff
@@ -33,59 +37,100 @@ where
     /// The opening argument.
     type Proof: Clone + Serialize + DeserializeOwned;
 
+    /// Possible proof verification errors.
     type Error: Debug;
 
-    /// This should return a coset domain (s.t. Domain::next_point returns Some)
+    /// This should return a domain such that `Domain::next_point` returns `Some`.
     fn natural_domain_for_degree(&self, degree: usize) -> Self::Domain;
 
+    /// Given a collection of evaluation matrices, produce a binding commitment to
+    /// the polynomials defined by those evaluations.
+    ///
+    /// Returns both the commitment which should be sent to the verifier
+    /// and the prover data which can be used to produce opening proofs.
     #[allow(clippy::type_complexity)]
     fn commit(
         &self,
         evaluations: Vec<(Self::Domain, RowMajorMatrix<Val<Self::Domain>>)>,
     ) -> (Self::Commitment, Self::ProverData);
 
+    /// Given prover data corresponding to a commitment of a collection of evaluation matrices,
+    /// return the evaluations of those matrices on the given domain.
+    ///
+    /// Ideally this should only be called when the domain is a subset of the evaluation domain
+    /// on which the evaluation matrices are defined. In these cases, this should be essentially
+    /// a no-op.
```

Comment:
"Ideally" is to vague for a comment; what happens if the assumption is not true?
```suggestion
    /// This is essentially a no-op when called with a `domain` which is a subset of the evaluation domain
    /// on which the evaluation matrices are defined.
```